x/pool-incentives/keeper/grpc_query.go

```diff
@@ -87,6 +87,23 @@ func (q Querier) IncentivizedPools(ctx context.Context, _ *types.QueryIncentiviz
 	lockableDurations := q.Keeper.GetLockableDurations(sdkCtx)
 	distrInfo := q.Keeper.GetDistrInfo(sdkCtx)
 
+	// Add epoch duration to lockable durations if not already present.
+	// This is to ensure that concentrated gauges (which run on epoch time) are
+	// always included in the query, even if the epoch duration changes in the future.
+	epochDuration := q.incentivesKeeper.GetEpochInfo(sdkCtx).Duration
+	epochAlreadyLockable := false
+	for _, lockableDuration := range lockableDurations {
+		if lockableDuration == epochDuration {
+			epochAlreadyLockable = true
+			break
+		}
+	}
+
+	// Ensure that we only add epoch duration if it does not already exist as a lockable duration.
+	if !epochAlreadyLockable {
+		lockableDurations = append(lockableDurations, epochDuration)
+	}
```

Comment:
On mainnet, epoch duration is 1 day, and we also have 1-day uptime. How are we going to differentiate between internal per-epoch gauges and 1 day uptime gauges?

Reply:
FYI, from @jonator 

```
Yeah we need to make sure it only refers to pools and gauges that are internally incentivized via mint incentives

For external incentives, we handle those in the opposite direction: by querying the gauges and linking those to pools

(Which, a separate topic, causes a lot of performance issues)

Like in hindsight, we could have reduced complexity by having the node return all gauges for a pool, internal and external
```